libsolidity/ast/Types.cpp

```diff
@@ -1660,7 +1660,9 @@ MemberList::MemberMap ContractType::nativeMembers(ContractDefinition const*) con
 				&it.second->declaration()
 			));
 	}
-	addNonConflictingAddressMembers(members);
+	// In 0.5.0 address members are not populated into the contract.
+	if (!_contract->sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::V050))
```

Comment:
Should we assert for `_contract != nullptr` or turn on 0.5.0 only if it was supplied?

Reply:
`_contract` is different from `m_contract`. The parameter only specifies the current scope, while `m_contract` is the actual `x` part in `x.member`.

Reply:
Cool, didn't notice the type has `m_contract`.

Reply:
Apparently `m_contract` doesn't behave correctly in this scenario for some reason.

Reply:
Oh I'm sorry. Of course it has to be the source unit where it is used in, not where it was defined.

Reply:
You initial comment was 100% correct.